lib/ffmpeg/transcoder.rb

```diff
@@ -145,5 +138,42 @@ def fix_encoding(output)
     rescue ArgumentError
       output.force_encoding("ISO-8859-1")
     end
+
+    def optimize_screenshot_parameters(options, transcoder_options)
```

Comment:
@d4rky-pl I do appreciate feedback and thanks. I broke up the method a bit and could have gone further, but think that the additional complexity is required given that options can be specified as an Array or Hash.